user/tests/wiring/no_fixture/thread.cpp

```diff
@@ -203,7 +203,13 @@ test(THREAD_07_particle_process_behavior_when_threading_enabled)
     assertTrue(Particle.connected());
 
     assertEqual((int)test_val_fn1, 0);
-    Particle.process();
+    // The increment task that was added to the Application Queue via invoke_async is not guaranteed to be the first and only item in the Application Thread Queue
+    // Call process() repeatedly in order to ensure that all potential messages in the queue are consumed, including the increment function
+    bool processed = false;
+    do {
+        processed = Particle.process();
```

Comment:
Nit: `while (Particle.process());` looks simpler? :)

And again, I'd probably add this to all the other tests where `Particle.process()` is used just in case, as I think the ones we outlined as having `Particle.process()` have been known to fail occasionally.

Reply:
Originally I was going to only execute a max of 20 times to avoid the possibility of an infinite loop, but I think the only way that can happen is if the async task being processed adds more item(s) back to the application queue. Ill make the change. 

Ill glance over the other tests that use `Particle.process()` and see if they need to be iterated on as well